salt/modules/logmod.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+'''
+On-demand logging
+=================
+
+.. versionadded:: Nitrogen
+
+The sole purpose of this module is logging messages in the (proxy) minion.
+It comes very handy when debugging complex Jinja templates, for example:
+
+.. code-block:: jinja
+
+    {%- for var in range(10) %}
+      {%- set logged = salt.log.info(var) %}
+    {%- endfor %}
+
+CLI Example:
+
+.. code-block:: bash
+
+    salt '*' log.error 'Please dont do that, this module is not for CLI use!'
```

Comment:
I agree on this, I have just pushed this commit: https://github.com/saltstack/salt/pull/39099/commits/fa2b6a63ee1ab6d289239c7f8450ad537904c100
I hope I did not misread @thatch45's comment & I'm looking forward to use this module! :)